assets/javascripts/discourse/components/ai-persona-editor.gjs

```diff
@@ -0,0 +1,236 @@
+        this.router.transitionTo(
+          "adminPlugins.discourse-ai.ai-personas.show",
+          this.args.model
+        );
+      } else {
+        this.toasts.show({
+          data: { message: I18n.t("discourse_ai.ai-persona.saved") },
+          duration: 2000,
+        });
+      }
+    } catch (e) {
+      this.args.model.setProperties(backupModel);
+      popupAjaxError(e);
+    } finally {
+      later(() => {
+        this.isSaving = false;
+      }, 1000);
+    }
+  }
+
+  @action
+  delete() {
+    return this.dialog.confirm({
+      message: I18n.t("discourse_ai.ai-persona.confirm_delete"),
+      didConfirm: () => {
+        return this.args.model.destroyRecord().then(() => {
+          this.args.personas.removeObject(this.args.model);
+          this.router.transitionTo(
+            "adminPlugins.discourse-ai.ai-personas.index"
+          );
+        });
+      },
+    });
+  }
+
+  @action
+  updateAllowedGroups(ids) {
+    this.editingModel.set("allowed_group_ids", ids);
+  }
+
+  @action
+  async toggleEnabled() {
+    this.args.model.set("enabled", !this.args.model.enabled);
+    if (!this.args.model.isNew) {
+      try {
+        await this.args.model.update({ enabled: this.args.model.enabled });
+      } catch (e) {
+        popupAjaxError(e);
+      }
+    }
+  }
+
+  @action
+  async togglePriority() {
+    this.args.model.set("priority", !this.args.model.priority);
+    if (!this.args.model.isNew) {
+      try {
+        await this.args.model.update({ priority: this.args.model.priority });
+
+        this.#sortPersonas();
+      } catch (e) {
+        popupAjaxError(e);
+      }
+    }
+  }
+
+  @action
+  showPriorityHelp() {
+    this.dialog.alert({
+      message: I18n.t("discourse_ai.ai-persona.priority_help"),
+    });
+  }
+
+  #sortPersonas() {
+    const sorted = this.args.personas.toArray().sort((a, b) => {
+      if (a.priority && !b.priority) {
+        return -1;
+      } else if (!a.priority && b.priority) {
+        return 1;
+      } else {
+        return a.name.localeCompare(b.name);
+      }
+    });
+    this.args.personas.clear();
+    this.args.personas.setObjects(sorted);
+  }
+
+  <template>
+    <form
+      class="form-horizontal ai-persona-editor"
+      {{didUpdate this.updateModel @model.id}}
+      {{didInsert this.updateModel @model.id}}
+      {{didInsert this.updateAllGroups @model.id}}
+    >
+      <div class="control-group">
+        <DToggleSwitch
+          class="ai-persona-editor__enabled"
+          @state={{@model.enabled}}
+          @label="discourse_ai.ai-persona.enabled"
+          {{on "click" this.toggleEnabled}}
+        />
+      </div>
+      <div class="control-group ai-persona-editor__priority">
+        <DToggleSwitch
+          class="ai-persona-editor__priority"
+          @state={{@model.priority}}
+          @label="discourse_ai.ai-persona.priority"
+          {{on "click" this.togglePriority}}
+        />
+        <a
```

Comment:
We can use `DTooltip` for this stuff now...I will push up a commit to change it.